include/mbedtls/ssl.h

```diff
@@ -535,6 +535,16 @@ typedef void mbedtls_ssl_set_timer_t( void * ctx,
  */
 typedef int mbedtls_ssl_get_timer_t( void * ctx );
 
+#if defined(MBEDTLS_X509_CRT_PARSE_C)
+/**
+ * \brief          Callback type: receive notification before X.509 chain
+ *                 building
+ *
+ * \param ctx      Context pointer
+ * \param crt      X.509 certificate pointer
+ */
+typedef void mbedtls_ssl_pre_verify_t( void *ctx, mbedtls_x509_crt *crt );
```

Comment:
The callback sees the peer certificate object but not the SSL connection object nor even the SSL configuration. From an implementation point of view, it could get a pointer to the SSL connection object. This would be more powerful, but more at risk of misuse. Open question: is it best to pass the SSL object, or to not pass it and discourage from accessing it?